pyvrp/cpp/Solution.cpp

```diff
@@ -180,23 +174,6 @@ Solution::Solution(ProblemData const &data, std::vector<Route> const &routes)
             throw std::runtime_error(msg.str());
         }
 
-    for (size_t client = 1; client <= data.numClients(); ++client)
-    {
-        if (data.client(client).required && visits[client] == 0)
-        {
-            std::ostringstream msg;
-            msg << "Client " << client << " is required but not present.";
-            throw std::runtime_error(msg.str());
-        }
-
-        if (visits[client] > 1)
-        {
-            std::ostringstream msg;
-            msg << "Client " << client << " is visited more than once.";
-            throw std::runtime_error(msg.str());
-        }
```

Comment:
We also had this check because it was really easy to do given that we wanted to check required clients. I don't think it's really worth having on its own, so I have removed `visits` tracking altogether.